packages/@aws-cdk/aws-stepfunctions/test/states-language.test.ts

```diff
@@ -614,6 +614,77 @@ describe('States Language', () => {
       .branch(state1.next(state2))
       .branch(state2)).toThrow();
+  }),
+
+  describe('findReachableStates', () => {
+
+    test('Can retrieve possible states from initial state', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const state1 = new stepfunctions.Pass(stack, 'State1');
+      const state2 = new stepfunctions.Pass(stack, 'State2');
+      const state3 = new stepfunctions.Pass(stack, 'State3');
+
+      const definition = state1
+        .next(state2)
+        .next(state3);
+
+      // WHEN
+      const states = stepfunctions.State.findReachableStates(definition.startState);
+
+      // THEN
+      expect(state1.toStateJson()).toStrictEqual(states[0].toStateJson());
+      expect(state2.toStateJson()).toStrictEqual(states[1].toStateJson());
+      expect(state3.toStateJson()).toStrictEqual(states[2].toStateJson());
+    });
+
+    test('Won\'t retrieve unreachable states', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const state1 = new stepfunctions.Pass(stack, 'State1');
+      const state2 = new stepfunctions.Pass(stack, 'State2');
+      const state3 = new stepfunctions.Pass(stack, 'State3');
+
+      state1.next(state2).next(state3);
+
+      // WHEN
+      const states = stepfunctions.State.findReachableStates(state2);
+
+      // THEN
+      expect(state2.toStateJson()).toStrictEqual(states[0].toStateJson());
+      expect(state3.toStateJson()).toStrictEqual(states[1].toStateJson());
+      expect(states.length).toStrictEqual(2);
+    });
+
+    test('Works with Choice and Parallel states', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const state1 = new stepfunctions.Choice(stack, 'MainChoice');
+      const stateCA = new stepfunctions.Pass(stack, 'StateA');
+      const stateCB = new stepfunctions.Pass(stack, 'StateB');
+      const statePA = new stepfunctions.Pass(stack, 'ParallelA');
+      const statePB = new stepfunctions.Pass(stack, 'ParallelB');
+      const state2 = new stepfunctions.Parallel(stack, 'RunParallel');
+      const state3 = new stepfunctions.Pass(stack, 'FinalState');
+      state2.branch(statePA);
+      state2.branch(statePB);
+      state1.when(stepfunctions.Condition.stringEquals('$.myInput', 'A' ), stateCA);
+      state1.when(stepfunctions.Condition.stringEquals('$.myInput', 'B'), stateCB);
+      stateCA.next(state2);
+      state2.next(state3);
+
+      const definition = state1.otherwise(stateCA);
+
+      // WHEN
+      const statesFromStateCB = stepfunctions.State.findReachableStates(stateCB);
+      const statesFromState1 = stepfunctions.State.findReachableStates(definition);
+
+      // THEN
+      expect(statesFromStateCB.length).toStrictEqual(1);
+      expect(statesFromState1.length).toStrictEqual(5);
+      expect(statesFromState1[0].toStateJson()).toStrictEqual(state1.toStateJson());
+      expect(statesFromState1[4].toStateJson()).toStrictEqual(state3.toStateJson());
```

Comment:
How about something like this?

```suggestion
      const expected = [state1, stateCA, stateCB, state2, state3];
      for (let i = 0; i < expected.length; i++) {
        expect(statesFromState1[i].toStateJson()).toStrictEqual(expected[i].toStateJson());
      }
```

Reply:
Done